sample-apps/discount-functions-sample-app/web/frontend/components/DiscountDetailsPage.jsx

```diff
@@ -74,45 +103,91 @@ export default function DiscountDetailsPage({
 
   return (
     <Page title="Details" breadcrumbs={[{ onAction: redirectToDiscounts }]}>
-      <Layout>
-        {errorMarkup}
-        <Layout.Section>
-          <Card>
-            <Card.Section>
-              <TextField
-                value={title}
-                onChange={setTitle}
-                label="Discount title"
-                autoComplete="on"
+      <Frame>
+        <Layout>
+          {successMarkup}
+          {errorMarkup}
+          <Layout.Section>
+            <MethodCard
+              title="Update discount"
+              discountClass={savedDiscount.discountClass}
+              discountTitle={{
+                value: title,
+                onChange: setTitle,
+              }}
+              discountCode={{
+                value: code,
+                onChange: setCode
+              }}
+              discountMethod={{
+                value: method
+              }}
+              discountMethodHidden={true}
+            />
+            <Card>
+              <Card.Section>
+                {renderConfigurationForm(configuration, setConfiguration)}
+              </Card.Section>
+            </Card>
+            {method === DiscountMethod.Code && (
+              <UsageLimitsCard
+                totalUsageLimit={{
+                  value: usageLimit,
+                  onChange: (value) => setUsageLimit(parseInt(value)),
```

Comment:
(we talked about this in slack but leaving it here too) This breaks the input, it expects a string. Suggest changing to `onChange: setUsageLimit` and casting to an int in the submit mutation